lib/phlex/rails/form.rb

```diff
@@ -5,63 +5,15 @@ module Rails
 		class Form < Phlex::HTML
 			def initialize(model)
 				@model = model
-			end
-
-			def self.input_field(method_name, type:)
-				define_method method_name do |field, value: @model.attributes[field.to_s], **attributes|
-					input(
-						name: field_name(field),
-						type: type,
-						value: value,
-						**attributes
-					)
-				end
+				@url = @_view_context.url_for(@model)
+				@method = @model.persisted? ? :patch : :post
```

Comment:
This is what I am doing for the`@method`:

```ruby
@method = form_options.delete(:method) do
  break :patch if object.respond_to?(:persisted?) && object.persisted?
  :post
end
```

We definitely can't ignore :put, but I think it makes sense just to both allow additional options on initialize and allow the method to be one of them.

Reply:
Good point. I'm thinking the solution here is for initialize to accept `**opts`. Then `url`, `method`, and any other configurable options willl become:
```
@url = opts[:url] || ...
@method = opts[:method] || model.persisted? ? :patch : :post
```